test/parallel/test-buffer-slice.js

```diff
@@ -28,80 +28,55 @@ assert.strictEqual(0, Buffer.from('hello', 'utf8').slice(0, 0).length);
+  [buf.slice(10), Buffer.from('', 'utf8')],
+  [buf.slice(5, 8), Buffer.from('567', 'utf8')],
+  [buf.slice(8, -1), Buffer.from('8', 'utf8')],
+  [buf.slice(-10), Buffer.from('0123456789', 'utf8')],
+  [buf.slice(0, -9), Buffer.from('0', 'utf8')],
+  [buf.slice(0, -10), Buffer.from('', 'utf8')],
+  [buf.slice(0, -1), Buffer.from('012345678', 'utf8')],
+  [buf.slice(2, -2), Buffer.from('234567', 'utf8')],
+  [buf.slice(0, 65536), Buffer.from('0123456789', 'utf8')],
+  [buf.slice(65536, 0), Buffer.from('', 'utf8')],
+  [buf.slice(-5, -8), Buffer.from('', 'utf8')],
+  [buf.slice(-5, -3), Buffer.from('56', 'utf8')],
+  [buf.slice(-10, 10), Buffer.from('0123456789', 'utf8')],
+  [buf.slice('0', '1'), Buffer.from('0', 'utf8')],
+  [buf.slice('-5', '10'), Buffer.from('56789', 'utf8')],
+  [buf.slice('-10', '10'), Buffer.from('0123456789', 'utf8')],
+  [buf.slice('-10', '-5'), Buffer.from('01234', 'utf8')],
+  [buf.slice('-10', '-0'), Buffer.from('', 'utf8')],
+  [buf.slice('111'), Buffer.from('', 'utf8')],
+  [buf.slice('0', '-111'), Buffer.from('', 'utf8')]
+];
 
-assert.strictEqual(0, Buffer.compare(buf.slice(2),
-                                     Buffer.from('23456789', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(5),
-                                     Buffer.from('56789', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(10),
-                                     Buffer.from('', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(5, 8),
-                                     Buffer.from('567', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(8, -1),
-                                     Buffer.from('8', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(-10),
-                                     Buffer.from('0123456789', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(0, -9),
-                                     Buffer.from('0', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(0, -10),
-                                     Buffer.from('', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(0, -1),
-                                     Buffer.from('012345678', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(2, -2),
-                                     Buffer.from('234567', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(0, 65536),
-                                     Buffer.from('0123456789', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(65536, 0),
-                                     Buffer.from('', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(-5, -8),
-                                     Buffer.from('', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(-5, -3),
-                                     Buffer.from('56', 'utf8')));
-assert.strictEqual(0, Buffer.compare(buf.slice(-10, 10),
-                                     Buffer.from('0123456789', 'utf8')));
 for (let i = 0, s = buf; i < buf.length; ++i) {
```

Comment:
I know this wasn't introduced by this change, but this segment of the test seems broken to me. `s = buf` means that they are pointing to the same buffer in memory, so of course `buf.slice(i)` and `s.slice(i)` are going to be the same thing. Maybe that's the point of the test? But I suspect not. Anyone understand this? Worth just removing this entire block? If not, it seems like it could use a comment.

Reply:
This segment was imported in [this commit](https://github.com/nodejs/node/commit/7053922c1a76ac56ef245923ec171b01cd90d2f4#diff-5588e86c18e70132808bc84388581211) by @jasnell a year ago. I'm also confused with it.

@jasnell Could you tell us something about it ?

Reply:
@starkwang I don't think that's the commit that introduced it. It's `s = buf.toString()` in that commit, which is very different from `s = buf`.

It appears that it was introduced in https://github.com/nodejs/node/commit/02a4c081e93e527c1cff163534364a8d488c3777 and that it was an error. Does it make sense to add a second commit here or else in a different PR to restore it to `buf.toString()`?

Reply:
@Trott I believe it should be `buf.toString()` as well, the `s` probably means "string" here (would be clearer if it gets renamed to `str`). Fixing it here SGTM.

Reply:
OK, I'll change it into `buf.toString()`.